applications/fishing-map/src/features/timebar/timebar.hooks.ts

```diff
@@ -1,30 +1,69 @@
 import { useDispatch, useSelector } from 'react-redux'
-import { useCallback, useEffect } from 'react'
+import { useCallback, useEffect, useMemo } from 'react'
+import { atom, useRecoilState } from 'recoil'
+import { debounce } from 'lodash'
+import { createSelector } from 'reselect'
 import { TimebarVisualisations } from 'types'
-import { selectTimeRange, selectTimebarVisualisation } from 'features/app/app.selectors'
+import { selectTimebarVisualisation } from 'features/app/app.selectors'
 import { useLocationConnect } from 'routes/routes.hook'
 import {
   selectActiveActivityDataviews,
   selectActiveTrackDataviews,
 } from 'features/dataviews/dataviews.selectors'
-import { setStaticTime, selectHasChangedSettingsOnce, changeSettings } from './timebar.slice'
+import { DEFAULT_TIME_RANGE } from 'data/config'
+import store, { RootState } from 'store'
+import { selectUrlTimeRange } from 'routes/routes.selectors'
+import { selectWorkspaceTimeRange } from 'features/workspace/workspace.selectors'
+import { selectHasChangedSettingsOnce, changeSettings, Range } from './timebar.slice'
 
-export const useTimerangeConnect = () => {
-  const { dispatchQueryParams } = useLocationConnect()
-  const dispatch = useDispatch()
-  const { start, end } = useSelector(selectTimeRange)
-  // TODO needs to be debounced like viewport
-  const dispatchTimeranges = useCallback(
-    (event: { start: string; end: string; source: string }) => {
-      const range = { start: event.start, end: event.end }
-      if (event.source !== 'ZOOM_OUT_MOVE') {
-        dispatch(setStaticTime(range))
+const selectTimeRange = createSelector(
```

Comment:
I'd keep this into the .selector file as it is a selector and having it within a .hook file could be confusing